ipa-core/src/secret_sharing/mod.rs

```diff
@@ -211,9 +240,39 @@ impl<F: Field> FieldSimd<1> for F {}
 
 impl FieldSimd<32> for Fp32BitPrime {}
 
-impl FieldSimd<64> for Boolean {}
+macro_rules! boolean_vector {
+    ($dim:expr, $vec:ty) => {
+        impl Vectorizable<$dim> for Boolean {
+            type Array = $vec;
+        }
+
+        impl FieldVectorizable<$dim> for Boolean {
+            type ArrayAlias = $vec;
+        }
+
+        impl FieldSimd<$dim> for Boolean {}
+
+        impl From<AdditiveShare<$vec>> for AdditiveShare<Boolean, $dim> {
+            fn from(value: AdditiveShare<$vec>) -> Self {
+                AdditiveShare::new_arr(value.left(), value.right())
+            }
+        }
+
+        impl From<AdditiveShare<Boolean, $dim>> for AdditiveShare<$vec> {
+            fn from(value: AdditiveShare<Boolean, $dim>) -> Self {
+                AdditiveShare::new(*value.left_arr(), *value.right_arr())
+            }
+        }
+    };
+}
 
-impl FieldSimd<256> for Boolean {}
+boolean_vector!(3, BA3);
+boolean_vector!(5, BA5);
+boolean_vector!(8, BA8);
+boolean_vector!(20, BA20);
+boolean_vector!(32, BA32);
+boolean_vector!(64, BA64);
+boolean_vector!(256, BA256);
```

Comment:
The set of sizes we generate values for isn't consistent.  I hope that is OK.  Will someone be surprised if they try to invoke something for which there is one implementation and not another?

Reply:
For vectorization in general, because rust's const generic support is limited, the supported vectorizations need to be enumerated explicitly in the source. I did the same thing here (listing only the ones that we use). Also, arbitrary vectorization dimensions are only needed where we're using the record-wise vectorization support to implement bit-wise vectorization within a record. Implementing only the necessary vectorizations helps to discourage that.